### ISSUE TITLE
Latest `wget` 1.19.3 produces errors

### ISSUE THREAD

**Issue by sdrausty** [termux/termux-packages#2052]
`wget https://github.com/sdrausty/TermuxArch/blob/master/setupTermuxArch.md5` produces strange output.  This is with the latest version of `wget`.  The previous version, 1.19.2 works well.  This causes https://sdrausty.github.io/TermuxArch/ to fail.  

> How to solve this error? 

*From https://github.com/sdrausty/TermuxArch/issues/35*

**Comment by xeffyr** [comment]
There also wget test suite failure:
```
FAIL: Test-https-tlsv1x.px
FAIL: Test-https-pfs.px
FAIL: Test-https-tlsv1.px
FAIL: Test-https-selfsigned.px
PASS: Test-https-badcerts.px
  CCLD     unit-tests
PASS: unit-tests
FAIL: Test-https-crl.px
FAIL: Test-https-weboftrust.px
FAIL: Test-https-clientcert.px
============================================================================
Testsuite summary for wget 1.19.3
============================================================================
# TOTAL: 93
# PASS:  72
# SKIP:  14
# XFAIL: 0
# FAIL:  7
# XPASS: 0
# ERROR: 0
============================================================================
See tests/test-suite.log
Please report to bug-wget@gnu.org
============================================================================
```

But it seems that fails only https.

**Comment by xeffyr** [comment]
I'm getting this problem even on desktop linux installation (ArchLinux, x86_64).
Wget corrupts file content:
```
~/testing2 ➤ LANG=en_US.UTF-8 wget https://raw.githubusercontent.com/sdrausty/TermuxArch/master/setupTermuxArch.md5
--2018-01-22 00:41:14--  https://raw.githubusercontent.com/sdrausty/TermuxArch/master/setupTermuxArch.md5
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving raw.githubusercontent.com... 151.101.0.133, 151.101.192.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 77 [text/plain]
Saving to: ‘setupTermuxArch.md5’

setupTermuxArch.md5                   100%[=======================================================================>]      77  --.-KB/s    in 0s      

2018-01-22 00:41:15 (59.3 MB/s) - ‘setupTermuxArch.md5’ saved [77/77]

~/testing2 ➤ cat setupTermuxArch.md5
�03H67H3K6J107L63I61HK4�062I140HSP(N-)I-�-�p,J��+I,�K���r�n9
```

While curl downloads this file without any problems:
```
~/testing2 ➤ curl https://raw.githubusercontent.com/sdrausty/TermuxArch/master/setupTermuxArch.md5                 
860c70f6c2d071c64c40fa08324d100f  setupTermuxArch.tar.gz
```

**Comment by xeffyr** [comment]
I have found that file is not getting corrupted, but gzipped 😃.

```
bash-4.4$ cat setupTermuxArch.md5
�L�465K117O5�4ML3��HK31��LN13L41TP(N-)I-�-�p,J��+I,�K���᧹9bash-4.4$ cat setupTermuxArch.md5 | gzip -d
9d9356d477e795af698ff4189cd61a41  setupTermuxArch.tar.gz
```

**Comment by sdrausty** [comment]
>  this problem exists only in certain cases.

@staymetal and I were ripping our heads apart over this yesterday. 
> this problem exists only in certain cases

Out of three different armv7 device tests yesterday, installation on two devices was successful. 

All armv8 device installs were successful too.

**Comment by xeffyr** [comment]
Wget 1.19.3 works without error if additional flag passed:
```
wget --compression=none -qO- https://raw.githubusercontent.com/sdrausty/TermuxArch/master/setupTermuxArch.md5
860c70f6c2d071c64c40fa08324d100f  setupTermuxArch.tar.gz
```

It seems that gzip decompression sometimes does not work.

**Comment by sdrausty** [comment]
A solution is to downgrade wget in Termux.  Use `apt remove wget` to downgrade your wget to https://sdrausty.github.io/TermuxArch/scripts/files/wget_1.19.2-1_aarch64.deb this version.  `dpkg --install wget_1.19.2-1_aarch64.deb` is a resolution this issue for the time being.

**Comment by sdrausty** [comment]
> Wget 1.19.3 works without error if additional flag passed:

It play a nice melody now through the vibrator https://github.com/sdrausty/TermuxArch/commit/ab020609020058e1dbdcbde4befdcb0ae02fd1c0 with the additional arguments.  

>  wget -q -N --show-progress https://raw.githubusercontent.com/sdrausty/TermuxArch/master/setupTermuxArch.md5   
wget -q -N --show-progress --compression=none -qO- https://raw.githubusercontent.com/sdrausty/TermuxArch/master/setupTermuxArch.md5

**Comment by sdrausty** [comment]
For armv7 users https://sdrausty.github.io/TermuxArch/scripts/files/wget_1.19.2_arm.deb

**Comment by sdrausty** [comment]
@xeffyr 
> not getting corrupted, but gzipped 

This file being corrupted through gzipping.

**Comment by SamSanuch** [comment]
@xeffyr What even in Linux when I a file from you s7 kernel from GitHub through wget it messed text inside file , I recommend to only use wget in downloading binaries and zips! And curl is better for text files!

**Comment by sdrausty** [comment]
Downgrading `wget` according to instructions has just closed https://github.com/sdrausty/TermuxArch/issues/35 --~~~~

**Comment by sdrausty** [comment]
https://github.com/sdrausty/TermuxArch/issues/32#issuecomment-359306018 `wget` might be producing other maladies as well with version 1.19.3.  

>  Automatic x86_64 download from evowise no longer works https://github.com/sdrausty/TermuxArch/issues/32 
How to solve this error?  https://github.com/sdrausty/TermuxArch/issues/35
Latest `wget` 1.19.3 produces errors https://github.com/termux/termux-packages/issues/2052
**All three(3) of these may be related, including this topic.**  

**All three(3) of these may be related, including this topic to wget 1.19.3 and libunistring.**  

> if [ $(getprop ro.product.cpu.abi) = x86_64 ];then
		wget -A tar.gz -m -nd -np http://$mirror$path

*From https://raw.githubusercontent.com/sdrausty/TermuxArch/master/scripts/files/stable/necessaryfunctions.sh*

**Comment by its-pointless** [comment]
whats happening is that its downloading a gzipped stream and not decompressing. 
wget a file rename it to file.gz
gunzip file.gz

**Comment by its-pointless** [comment]
http://git.savannah.gnu.org/cgit/wget.git/patch/?id=047746eb765d212cbf05afac16f4bd73257d8186 
revert that and it works ...

**Comment by sdrausty** [comment]
@its-pointless thank you for your comments.  

`setupTermuxArch.tar.gz 100%[==========================>]   6.87K  --.-KB/s    in 0.05s
wget: unrecognized option '--without-zlib'
Usage: wget [OPTION]... [URL]...

Try `wget --help' for more options.
[23:49 t ]$ wget --version
GNU Wget 1.19.2 built on linux-androideabi.`

**Comment by sdrausty** [comment]
`wget -q -N --show-progress https://raw.githubusercontent.com/sdrausty/TermuxArch/master$flag/setupTermuxArch.md5 |xargs gunzip`

```
setupTermuxArch.tar.gz 100%[==========================>]   6.87K  --.-KB/s    in 0.06s
setupTermuxArch.md5    100%[==========================>]      77  --.-KB/s    in 0s
gunzip: invalid magic
```

**Comment by sdrausty** [comment]
> recommend to only use wget in downloading binaries and zips! And curl is better for text files

Thank you for your comment @SamSanuch wget version 1.19.2 is good at both.  Version 1.19.3 obviously is not toi good a both types of downloads.

**Comment by fornwall** [comment]
Thanks - this has been fixed in the updated version 1.19.4 of wget which is now available as an update!

**Comment by sdrausty** [comment]
@fornwall Thank you for the quick update!    

```
$ wget --version
GNU Wget 1.19.4 built on linux-androideabi.
```
Works as expected.
